src_c/methods.c

```diff
@@ -0,0 +1,102 @@
+#include "include/pygame.h"
+#include "include/geometry.h"
+#include "include/collisions.h"
+#include "include/methods.h"
+
+#include <limits.h>
+#include <float.h>
+#include <math.h>
+
+#ifndef PySequence_FAST_CHECK
+#define PySequence_FAST_CHECK(o) (PyList_Check(o) || PyTuple_Check(o))
+#endif /* ~PySequence_FAST_CHECK */
+
+static PyObject *
+pg_geometry_raycast(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    PyObject **A_farr;
+    PyObject **B_farr;
+    Py_ssize_t A_len;
+    Py_ssize_t B_len;
+    Py_ssize_t i;
+    Py_ssize_t j;
+
+    if (nargs != 2) {
+        return RAISE(PyExc_TypeError, "raycast() takes exactly 2 arguments");
+    }
+    if (!PySequence_FAST_CHECK(args[0])) {
+        return RAISE(PyExc_TypeError,
+                     "raycast() first argument must be a sequence");
+    }
+    if (!PySequence_FAST_CHECK(args[1])) {
+        return RAISE(PyExc_TypeError,
+                     "raycast() second argument must be a sequence");
+    }
+
+    A_len = PySequence_Fast_GET_SIZE(args[0]);
+    B_len = PySequence_Fast_GET_SIZE(args[1]);
+
+    if (A_len == 0) {
+        return PyList_New(0);
+    }
+    if (B_len == 0) {
+        PyObject *ret = PyList_New(A_len);
+        for (i = 0; i < A_len; i++) {
+            PyList_SET_ITEM(ret, i, Py_None);
+        }
```

Comment:
This can segfault, `Py_None` is a singleton and needs to be treated line any other object. `PyList_SET_ITEM` steals a reference to the object to be appended, so you are decreasing the Py_None refcount on each append (No good).
instead you should do the following
```suggestion
        for (i = 0; i < A_len; i++) {
            Py_INCREF(Py_None);
            PyList_SET_ITEM(ret, i, Py_None);
        }
```